src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,813 @@
+         
+         auto dragArea = safenew wxStaticBitmap(this, wxID_ANY, theTheme.Bitmap(bmpDragArea));
+         dragArea->Disable();
+         sizer->Add(
+            dragArea,
+            wxSizerFlags().Center().Border(wxLEFT, 5)
+         );
+         sizer->Add(
+            mEnableButton,
+            wxSizerFlags().Center().Border(wxLEFT, 5)
+         );
+         sizer->Add(
+            mChangeButton,
+            wxSizerFlags().Center().Border(wxLEFT, 5).Proportion(1)
+         );
+         sizer->Add(
+            optionsButton,
+            wxSizerFlags().Center().Border(wxLEFT | wxRIGHT, 5)
+         );
+
+         SetSizer(sizer.release());
+      }
+
+      void SetEffect(const std::shared_ptr<Track>& track, RealtimeEffectState* state)
+      {
+         mTrack = track;
+         mEffectState = state;
+         auto desc = PluginManager::Get().GetPlugin(mEffectState->GetID());
+
+         mChangeButton->SetTranslatableLabel(desc->GetSymbol().Msgid());
+      }
+
+      void RemoveFromList()
+      {
+         if(mEffectState != nullptr)
+         {
+            auto project = mTrack->GetOwner()->GetOwner();
+            auto effectName = mEffectState->GetEffect()->GetName();
+            AudioIO::Get()->RemoveState(*project, &*mTrack, *mEffectState);
+            ProjectHistory::Get( *project ).PushState(
+               //i18n-hint: undo history, first parameter - realtime effect name, second - track name
+               XO("'%s' removed from '%s' effect stack").Format(effectName, mTrack->GetName()),
+               //i18n-hint: undo history record
+               XO("Remove realtime effect")
+            );
+         }
+      }
+      
+      void OnOptionsClicked(wxCommandEvent& event)
+      {
+         if(mEffectState == nullptr)
+            return;//not initialized
+         
+         auto effectPlugin = EffectManager::Get().GetEffect(mEffectState->GetID());
+         if(effectPlugin == nullptr)
+            //Effect was deleted/disabled?
+            return;
+
+         auto trackList = mTrack->GetOwner();
+         if(!trackList)
+            return;
+
+         auto project = trackList->GetOwner();
+         if(project == nullptr)
+            return;
+
+         auto access = mEffectState->GetAccess();
+            
+         access->ModifySettings([&](EffectSettings& settings)
+         {
+            effectPlugin->DoEffect(settings,
+               ProjectRate::Get(*project).GetRate(),
+               &*mTrack->GetOwner(),
+               &WaveTrackFactory::Get( *project ),
+               ViewInfo::Get( *project ).selectedRegion,
+               EffectManager::kNone,
+               &ProjectWindow::Get( *project ),
+               EffectUI::DialogFactory,
+               access);
+         });
+      }
+
+      void OnChangeButtonClicked(wxCommandEvent& event)
+      {
+         if(mEffectState == nullptr)
+            return;//not initialized
+
+         ShowSelectEffectMenu(mChangeButton, this);
```

Comment:
I see that this call pops a menu and returns a plugin id which is ignored.  So it's still to do:  substitute one effect for another in the list, and also push an undo item.

This PR might be merged with this known incompleteness.  (Should have a TODO comment)